### PR TITLE
chore(docs): updated the name of Twitter to X

### LINKED ISSUE #3065: Update the name of Twitter to X in documentations
With everything evolving so rapidly, wouldn't it be best to stay updated with the latest?
Everyone must be well aware by now that the legal name of Twitter has been formally changed to X. Therefore, the issue addresses the problem of improving the doc files and updating the name to X.

@mrgrain 
Kindly assign me this issue. I would like to work on this issue as I am new to open-source.

### PR DESCRIPTION
**Pull request by tvermaashutosh**
Fixes #3065 

Everyone must be well aware by now that the legal name of Twitter has been formally changed to X.
In this PR I have updated the name of Twitter to X in all the documentation (Texts and Links) of this repo.
Closes: https://github.com/reworkd/AgentGPT/issues/1345

@mrgrain 
Kindly review my PR and merge it please.

---
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by mrgrain** [comment]
The contributor profiles are auto generated. We can't change them

File: src/cdk/construct-lib.ts [hunk 1]
```diff
@@ -23,7 +23,7 @@ export interface ConstructLibraryOptions extends JsiiProjectOptions {
 
 export interface Catalog {
   /**
-   * Twitter account to @mention in announcement tweet.
+   * X account to @mention in announcement tweet.
```
Comment (mrgrain) [inline src/cdk/construct-lib.ts:26]: Can you add something like (previously Twitter) to explain the difference in docs and property.

File: src/cdk/construct-lib.ts [hunk 2]
```diff
@@ -7,11 +7,11 @@ export interface ConstructLibraryOptions extends JsiiProjectOptions {
    *
    *     https://awscdk.io/packages/[@SCOPE/]PACKAGE@VERSION
    *
-   * The catalog will also post a tweet to https://twitter.com/awscdkio with the
+   * The catalog will also post a tweet to https://x.com/awscdkio with the
    * package name, description and the above link. You can disable these tweets
    * through `{ announce: false }`.
    *
-   * You can also add a Twitter handle through `{ twitter: 'xx' }` which will be
+   * You can also add a X handle through `{ X: 'xx' }` which will be
```
Comment (mrgrain) [inline src/cdk/construct-lib.ts:14]:
```suggestion
   * You can also add a X handle through `{ twitter: 'xx' }` which will be
```
This is an external API the can't changed unilaterally from here.

File: src/cdk/construct-lib.ts [hunk 3]
```diff
@@ -7,11 +7,11 @@ export interface ConstructLibraryOptions extends JsiiProjectOptions {
    *
    *     https://awscdk.io/packages/[@SCOPE/]PACKAGE@VERSION
    *
-   * The catalog will also post a tweet to https://twitter.com/awscdkio with the
+   * The catalog will also post a tweet to https://x.com/awscdkio with the
```
Comment (mrgrain) [inline src/cdk/construct-lib.ts:10]:
```suggestion
   * The catalog will also post to https://x.com/awscdkio with the
```

**Review by mrgrain — CHANGES_REQUESTED** [review]

File: src/cdk/construct-lib.ts (hunk 3 above)
Comment (tvermaashutosh) [reply src/cdk/construct-lib.ts:10]: Just wait a minute. I'm making the required changes.

**Comment by tvermaashutosh** [comment]
@mrgrain 
I have made all the requested changes in the PR.
Kindly review it now and merge it please.

File: README.md [hunk 4]
```diff
@@ -404,7 +404,7 @@ Thanks goes to these wonderful people ([emoji key](https://allcontributors.org/d
...
       <td align="center" valign="top" width="14.28%"><a href="https://github.com/jmourelos"><img src="https://avatars3.githubusercontent.com/u/3878434?v=4?s=100" width="100px;" alt="Jacob"/><br /><sub><b>Jacob</b></sub></a><br /><a href="https://github.com/projen/projen/commits?author=jmourelos" title="Code">💻</a></td>
       <td align="center" valign="top" width="14.28%"><a href="http://joapy.com/"><img src="https://avatars3.githubusercontent.com/u/325306?v=4?s=100" width="100px;" alt="Jake Pearson"/><br /><sub><b>Jake Pearson</b></sub></a><br /><a href="https://github.com/projen/projen/commits?author=jakepearson" title="Code">💻</a></td>
-      <td align="center" valign="top" width="14.28%"><a href="http://twitter.com/bracki"><img src="https://avatars.githubusercontent.com/u/49786?v=4?s=100" width="100px;" alt="Jan Brauer"/><br /><sub><b>Jan Brauer</b></sub></a><br /><a href="https://github.com/projen/projen/commits?author=bracki" title="Code">💻</a></td>
```
Comment (mrgrain) [inline README.md:407]: Needs reverting

File: docs/api/API.md [hunk 5]
```diff
@@ -15392,7 +15392,7 @@ Name | Type | Description
 Name | Type | Description 
 -----|------|-------------
 **announce**?🔹 | <code>boolean</code> | Should we announce new versions?<br/>__*Default*__: true
-**twitter**?🔹 | <code>string</code> | Twitter account to.<br/>__*Optional*__
+**X**?🔹 | <code>string</code> | X account to.<br/>__*Optional*__
```
Comment (mrgrain) [inline docs/api/API.md:15395]: This file shouldn't be changed manually, instead run the build. If you have to, see the build for the proper change.

**Review by mrgrain — CHANGES_REQUESTED** [review]

**Review by mrgrain — CHANGES_REQUESTED** [review]
Some more stuff. Also the PR title needs fixing.

**Comment by tvermaashutosh** [comment]
@mrgrain
I have made all the requested changes in the PR.
Kindly review it now and merge it please.

**Comment by mrgrain** [comment]
Build is still failing.

**Comment by mrgrain** [comment]
@tvermaashutosh Are you still interested in finishing this change?

**Pull request closed without merging**